File: src/bin/isset.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, char *argv[]) {
    int verbose = 0;

    if (argc < 2) {
        return 1;
    }

    if (argc > 2 && strcmp(argv[1], "-v") == 0) {
        verbose = 1;
    }

    if (argc > 2 + verbose) {
        printf("Only one variable is checkable at the same time!");
        return 2;
    }

    char *key = argv[1 + verbose];
    char *value = getenv(key);

    if (value != NULL && verbose == 1) {
        printf("%s is set with value = %s", key, value);
    } else if (value != NULL && verbose == 0){
        printf("Given variable '%s' is set.", key);
    } else {
        printf("Given variable '%s' is not set.", key);
    }

    return value == NULL;
}

File: src/bin/isset.rs
use std::env;
use std::process;

fn main(){

    let args: Vec<String>= env::args().collect();
    let mut verbose = false;

    if args.len() < 2 {
        println!("No parameter was given!");
        process::exit(1);
    }

    if args.len() > 2 && args[1] == "-v" {
        verbose = true;
    }

    if args.len() > (2 + verbose as usize) {
        println!("Only one variable is checkable at a time!");
        process::exit(2);
    }


    // key is either args[1] (no verbose) or args[2] (with verbose)
    let key_index = 1 + verbose as usize;
    let key = &args[key_index];
    let value = env::var(key).ok();

    match(value, verbose){
        (Some(v), true) => {
            println!("{} is set with value = {}", key, v);
            process::exit(0);
        }
        (Some(_), false) => {
            println!("Given variable '{}' is set.", key);
            process::exit(0);
        }
        (None, _) => {
            println!("Given variable '{}' is not set.", key);
            process::exit(1);
        }
    }
}